kv/kvserver: TestReplicateQueueRebalanceMultiStore failed

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/4352ff8f-3cde-4485-8eba-b9d931461e82?testReportRun=3&testReportShard=27&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [abe3d27bee5120afa287df2f8cc81c82ab6864da](https://github.com/cockroachdb/cockroach/commits/abe3d27bee5120afa287df2f8cc81c82ab6864da):

Fatal error:

```
panic: test timed out after 39m57s
running tests:
	TestReplicateQueueRebalanceMultiStore (39m28s)
	TestReplicateQueueRebalanceMultiStore/multi-store (30m27s)

```
Stack: 

```
goroutine 197105750 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicateQueueRebalanceMultiStore
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestReplicateQueueRebalanceMultiStore3392336486
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestReplicateQueueRebalanceMultiStore/multi-store
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=27</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41298

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -182,6 +182,7 @@ func TestReplicateQueueRebalanceMultiStore(t *testing.T) {
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t)
 	skip.UnderShort(t)
+	skip.UnderDeadlock(t)
 
 	testCases := []struct {
 		name          string

